libnmstate/netinfo.py

```diff
@@ -30,43 +31,66 @@ def show():
 
 
 def interfaces():
-    return {
-        'interfaces': [
-            {
-                'name': dev['name'],
-                'type': dev['type'],
-                'state': dev['state'],
-            }
-            for dev in devices()
-        ]
-    }
+    info = {'interfaces': []}
+    for dev in devices():
+        iface_info = {
+            'name': dev['name'],
+            'type': dev['type'],
+            'state': dev['state'],
+        }
+        # TODO: What about unmanaged devices?
+        if iface_info['type'] == 'bond' and 'link-aggregation' in dev:
+            iface_info['link-aggregation'] = dev['link-aggregation']
 
+        info['interfaces'].append(iface_info)
 
-def devices():
-    client = nmclient.client()
+    return info
 
-    devs = client.get_devices()
 
-    devlist = [
-        {
-            'name': dev.get_iface(),
-            'type_id': dev.get_device_type(),
-            'type': resolve_nm_dev_type(dev.get_type_description()),
-            'state': resolve_nm_dev_state(dev.get_state()),
-        }
-        for dev in devs
-    ]
+def devices():
+    devlist = []
+    for dev in nm.device.list_devices():
+        devinfo = _devinfo_common(dev)
+        devinfo.update(_devinfo_bond(dev, devinfo))
```

Comment:
IMHO if there are several methods that return the same object that they get as an parameter, it makes more sense to put them into a class.

Reply:
In this specific case, it implies that all info generators must be placed under the same "roof" and I see no good enough reason why that is needed.
I hope to see each type register to netinfo, with the relevant handler, making it simple for an additional type to be added without touching the core code.